src/argilla/client/feedback/metrics/annotator_metrics.py

```diff
@@ -0,0 +1,373 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""This module contains metrics to compare Annotator's suggestions vs responses. """
+
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Dict, List, Tuple, Union
+
+import numpy as np
+from pydantic import BaseModel
+
+from argilla.client.feedback.metrics.utils import get_responses_per_user
+from argilla.client.feedback.schemas import (
+    LabelQuestion,
+    MultiLabelQuestion,
+    RankingQuestion,
+    RatingQuestion,
+    TextQuestion,
+)
+from argilla.utils.dependency import requires_dependencies
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset import FeedbackDataset
+
+import pandas as pd
+from sklearn.metrics import (
+    accuracy_score,
+    confusion_matrix,
+    f1_score,
+    multilabel_confusion_matrix,
+    precision_score,
+    recall_score,
+)
+
+
+class AnnotatorMetricResult(BaseModel):
+    """Container for the result of an annotator metric.
+
+    It contains two fields, `metric_name` and `result` with the value of the metric.
+    """
+
+    metric_name: str
+    result: Union[float, Dict[str, float], pd.DataFrame, List[pd.DataFrame]]
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class AnnotatorMetric:
+    def __init__(self, dataset: "FeedbackDataset", question_name: str) -> None:
+        """Initializes a `AnnotatorMetric` object to compute metrics on
+        a `FeedbackDataset` for a given question at the annotator level.
+
+        Args:
+            dataset: FeedbackDataset to compute the metrics.
+            question_name: Name of the question for which we want to analyse the analyze
+                the annotator responses vs the suggestions given.
+
+        Raises:
+            NotImplementedError: If the question type is not supported.
+        """
+        self._dataset = dataset
+        self._question_name = question_name
+        self._question_type = type(self._dataset.question_by_name(question_name))
+        if self._question_type in (MultiLabelQuestion, RankingQuestion):
+            raise NotImplementedError(f"No metrics are defined currently for {self._question_type.__name__}")
+        self._allowed_metrics = METRICS_PER_QUESTION[self._question_type]
+
+    def compute(self, metric_names: Union[str, List[str]], **kwargs) -> Dict[str, List[AnnotatorMetricResult]]:
+        """Computes the annotator metrics for the given question.
+
+        Args:
+            metric_names: name or list of names for the metrics to compute. i.e. `alpha`
+            kwargs: additional arguments to pass to the metric.
```

Comment:
are these going to be the same? Otherwise, we might want to have a custom method per metric or a `compute`that just works for a single metric? Not sure, but looping through multiple metrics might mix/use the wrong **kwargs for other metrics?

Reply:
I'm thinking that too... passing a list of metrics is only expected to work if `kwargs` is not used. This applies just the same to the agreement metrics equivalent (but we only have one metric there for the moment). For the current implementation I think we don't need the kwargs, maybe we should just remove that possibility for the moment as it's simpler

Reply:
Yes, I think it might be better to remove `**kwargs` if we don't use it in order to avoid confusion